src/Adapter/Product/QueryHandler/GetProductCombinationsForEditingHandler.php

```diff
@@ -0,0 +1,136 @@
+{
+    /**
+     * @var CombinationDataProvider
+     */
+    private $combinationDataProvider;
+
+    /**
+     * @param CombinationDataProvider $combinationDataProvider
+     */
+    public function __construct(CombinationDataProvider $combinationDataProvider)
+    {
+        $this->combinationDataProvider = $combinationDataProvider;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(GetProductCombinationsForEditing $query): ProductCombinationsForEditing
+    {
+        $product = $this->getProduct($query->getProductId());
+        $combinationsInfo = $this->combinationDataProvider->getCombinationsInfo((int) $product->id, $query->getLimit(), $query->getOffset());
+
+        $combinationIds = array_map(function ($combination): int {
+            return (int) $combination['id_product_attribute'];
+        }, $combinationsInfo['combinations']);
+
+        $attributesInformation = $this->combinationDataProvider->getAttributesInfoByCombinationIds(
+            $combinationIds,
+            $query->getLanguageId()
+        );
+
+        return $this->formatCombinationsForEditing(
+            $combinationsInfo['combinations'],
+            $attributesInformation,
+            $combinationsInfo['total_count']
+        );
+    }
+
+    /**
+     * @param array $combinations
+     * @param array<int, CombinationForEditing> $attributesInformationByCombinationId
+     * @param int $totalCombinationsCount
+     *
+     * @return ProductCombinationsForEditing
+     */
+    private function formatCombinationsForEditing(
+        array $combinations,
+        array $attributesInformationByCombinationId,
+        int $totalCombinationsCount
+    ): ProductCombinationsForEditing {
+        $combinationsForEditing = [];
+
+        foreach ($combinations as $combination) {
+            $combinationId = (int) $combination['id_product_attribute'];
+            $combinationAttributesInformation = [];
+
+            foreach ($attributesInformationByCombinationId[$combinationId] as $attributesInfo) {
+                $combinationAttributesInformation[] = new CombinationAttributeInformation(
+                    (int) $attributesInfo['id_attribute_group'],
+                    $attributesInfo['attribute_group_name'],
+                    (int) $attributesInfo['id_attribute'],
+                    $attributesInfo['attribute_name']
+                );
+            }
+
+            $combinationsForEditing[] = new CombinationForEditing(
+                $combinationId,
+                $this->buildCombinationName($combinationAttributesInformation),
+                $combinationAttributesInformation
+            );
+        }
+
+        return new ProductCombinationsForEditing($totalCombinationsCount, $combinationsForEditing);
+    }
+
+    /**
+     * @param CombinationAttributeInformation[] $attributesInformation
+     *
+     * @return string
+     */
+    private function buildCombinationName(array $attributesInformation): string
```

Comment:
We don't need the product name before the attribute list? Or is it concatenated after?

Reply:
hmm, its only the combination name like in Bo product page:
![Screenshot from 2020-08-14 11-43-32](https://user-images.githubusercontent.com/31609858/90231199-78323780-de23-11ea-8349-56d3efe5eca5.png)